src/posix/platform/system.cpp

```diff
@@ -152,11 +155,18 @@ otInstance *otSysInit(otPlatformConfig *aPlatformConfig)
 #if OPENTHREAD_POSIX_CONFIG_DAEMON_ENABLE
     ot::Posix::Daemon::Get().Enable(instance);
 #endif
-    return instance;
+
+#if OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE && OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE
+    otBorderAgentStart(instance);
+#endif
+    return sInstance = instance;
 }
 
 void otSysDeinit(void)
 {
+#if OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE && OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE
+    otBorderAgentStop(sInstance);
+#endif
```

Comment:
```suggestion
void otSysDeinit(otInstance *aInstance)
{
#if OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE && OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE
    otBorderAgentStop(aInstance);
#endif
```

Reply:
Doing so will break the build of ot-br-posix. Any shall we introduce a new API like `otSysDeinitForInstance(otInstance *)`?

Reply:
Resolved via offline discussion.